sim.c

```diff
@@ -633,12 +653,14 @@ static void move_atom(struct board *board, atom *a, struct vector position)
         .position = position,
     };
     remove_atom(board, (struct atom_ref_at_position){ a, position });
+    *a = (*a & ~MOVEMENT_INDEX) | (board->movements.length & MOVEMENT_INDEX) | BEING_MOVED;
```

Comment:
This is going to overwrite stuff if there's overlap, right?  I guess if there's overlap, there will already be a collision, so it doesn't matter?  In any case, it might make sense to check that `*a & REMOVED` as well as that `board->movements.length` is less than 2^24.